src/component/button/HsiButton/HsiButton.tsx

```diff
@@ -181,17 +200,35 @@ export class HsiButton extends React.Component<HsiButtonProps, HsiButtonStatePro
             }
           );
 
-      featureInfoUrls.push(featureInfoUrl);
-
+      if (featureInfoUrl.indexOf('geoserver.action') >= 0) {
+        const ftName: string = layer.getSource().getParams().LAYERS;
```

Comment:
```suggestion
        const ftName: string = layer.getSource()?.getParams()?.LAYERS;
```

Null checks for source / params should be included here. Plz check if this suggestion is sufficient

Reply:
check is already given though filterFunction `layerFilter`